core/src/main/java/ai/timefold/solver/core/impl/heuristic/selector/move/factory/LegacyIteratorAdapter.java

```diff
@@ -0,0 +1,24 @@
+package ai.timefold.solver.core.impl.heuristic.selector.move.factory;
+
+import java.util.Iterator;
+
+import ai.timefold.solver.core.api.move.Move;
+
+final class LegacyIteratorAdapter<Solution_> implements Iterator<Move<Solution_>> {
+
+    private final Iterator<ai.timefold.solver.core.impl.heuristic.move.Move<Solution_>> moveIterator;
+
+    public LegacyIteratorAdapter(Iterator<ai.timefold.solver.core.impl.heuristic.move.Move<Solution_>> moveIterator) {
+        this.moveIterator = moveIterator;
+    }
+
+    @Override
+    public boolean hasNext() {
+        return moveIterator.hasNext();
+    }
+
+    @Override
+    public Move<Solution_> next() {
+        return new ai.timefold.solver.core.impl.heuristic.move.LegacyMoveAdapter<>(moveIterator.next());
```

Comment:
Is there a reason for not importing the class?

Reply:
We now have two `Move`s - one public, the other not. Unfortunately, I cannot import both. So I chose to import the public one, and the private one uses FQNs.

Reply:
Do we have two `LegacyMoveAdapter` classes?

Reply:
No. Because the legacy move is there to convert the old move to the new move.
The reverse operation is not necessary.

Reply:
I was referring to importing the class `ai.timefold.solver.core.impl.heuristic.move.LegacyMoveAdapter`.